feat(spec,tools): add log-sanitize

karlowich opened the pull request:
Adaptation of #526 and closes #525

Thanks @vikramkhurana-vigoursoft

Commit:
feat(spec,tools): add log-sanitize

Signed-off-by: Karl Bonde Torp <k.torp@samsung.com>

karlowich commented:
Closes #518

NaddiNadja commented on cijoe/tests/tools/test_xnvme.py:
```diff
@@ -79,10 +79,18 @@ def test_format(cijoe, device, be_opts, cli_args):
+        pytest.skip(reason="[be=ramdisk] does not implement sanitize")
+
+    pytest.skip(reason="QEMU NVMe devices doesn't support sanitize")
+
+    err, _ = cijoe.run(f"xnvme sanitize --sanact 0x2 {cli_args}")
+
+    assert not err
 
-    pytest.skip(reason="TODO: always fails. Investigate.")
+    # Get log sanitize status log page
+    err, _ = cijoe.run(f"xnvme log-sanitize {cli_args} --nsid {device['nsid']}")
```
Is test_sanitize still always skipped? And what if the test is run on non-QEMU devices?

karlowich replied:
Right, perhaps it would be better for the test to need a device with label `sanitize`.

mbrsamsung reviewed (dismissed):
lgtm

NaddiNadja commented on cijoe/tests/tools/test_xnvme.py:
```diff
@@ -75,14 +75,20 @@ def test_format(cijoe, device, be_opts, cli_args):
 def test_sanitize(cijoe, device, be_opts, cli_args):
     if be_opts["be"] == "linux" and be_opts["admin"] in ["block"]:
         pytest.skip(reason="[admin=block] does not implement sanitize")
+    if "ramdisk" in device["labels"]:
+        pytest.skip(reason="[be=ramdisk] does not implement sanitize")
```
Maybe stupid question, but why is the reason "[be=ramdisk] (...)", when this is not the condition we are looking for? Should the if-statement be `if be_opts["be"] == "ramdisk"`, or should the reason maybe be changed? 
(and also, is ramdisk both as a label *and* a backend?)

karlowich replied:
You are right, this has been fixed now

NaddiNadja reviewed (commented):
single question

karlowich committed:
fix(cijoe/xnvme): check for be=ramdisk

Previously we checked for ramdisk in the device labels, this was
misleading, since the reason for the skip says that the be=ramdisk is
is the problem.

Signed-off-by: Karl Bonde Torp <k.torp@samsung.com>

NaddiNadja reviewed (approved):
nice lgtm :)

safl reviewed (approved):
lgtm, made a note of a potential issue here: https://github.com/xnvme/xnvme/issues/578
But no reason to keep this back.
